### DISCUSSION TITLE
Can ASReview label all the articles automatically and are there variables for accuracy

### DISCUSSION THREAD

**Discussion by DLiangUU** [asreview/asreview#1257] [Q&A] [answered]
Hi Everyone, 

I am a PhD student and I want to use ASReview for my systematic review. In this relation, I have two questions regarding the tool that I hope someone can help me with. 

I'm looking to have approximately 50 000 articles at start. I'm therefore wondering if it is possible to use ASReview more as a prescreening tool to significantly reduce the number of articles for manual screening. In other words, if it is possible to use ASReview to screen e.g., 100 articles/ or until ASReview can make a good model for the articles. Then after this process have ASReview label all the articles as either included or not included. I tried to look into the documentation but could only find that ASReview was ranking the articles based on which is most likely to be included. 

Apart from this, are there any variables in the ASReview tool during the screening process that gives an estimate of how accurate the tool predicts relevant articles, or how accurate the tool thinks it predicts correctly? 

Thank you all very much in advance for your time and help!

**Comment by gimoAI** [comment, marked as answer]
Hi @DLiangUU,

The type of machine learning application you refer to requires a human-in-the-loop (HITL) method, where the goal is to output an accurate model in as few steps as possible, which can then be used to label unseen documents.

However, because systematic reviewing requires all relevant records to be seen by the researcher and the model needs to learn a diverse range of concepts, the term researcher-in-the-loop (RITL) was introduced which has three unique components compared to HITL (https://www.nature.com/articles/s42256-020-00287-7):

1. The primary output of the process is a selection of the records, not a trained machine learning model.
2. All records in the relevant selection are seen by a human at the end of the process.
3. The use-case requires a reproducible workflow and complete transparency is required.

> I tried to look into the documentation but could only find that ASReview was ranking the articles based on which is most likely to be included.

So this is correct, ASReview will learn from your screening decisions. After each screening decision it will update the ranking to present you the most likely relevant article first for screening (assuming you have the query strategy set to maximum, which is the default).

As of your second question:
> Apart from this, are there any variables in the ASReview tool during the screening process that gives an estimate of how accurate the tool predicts relevant articles, or how accurate the tool thinks it predicts correctly?

ASReview has no such variables, because it cannot know how many relevant articles are left to screen. Determining when you have screened enough and found most relevant articles is difficult, this topic is also described in the "How to stop?" section in the [active learning explained blogpost](https://asreview.nl/blog/active-learning-explained/).

Does this answer your questions?

**Reply by DLiangUU** [reply]
Dear GimoAI,

Thank you very much for the quick reply! 
So if I understand correctly. What I want to do, i.e., labeling all the records ASReview thinks are relevant, is outside of the scope of what the tool can do, as the goal of the tool is the ranking of relevancy of the records?

Additionally, for the second question, there is no variable for the accuracy of ASReview for the prediction of relevant records. What about during screening? Is there any form of measures that can track the rankings and how many of the articles ranked highly relevant are actually included? I know there is an overall tracker of how many articles have been included/excluded and a number of irrelevant since the last relevant but is there some way to also track e.g., the number of articles in a row that has been included by the researcher?

Thank you very much for your time and also for the two resources you have provided!

**Reply by gimoAI** [reply]
Hi,

>  labeling all the records ASReview thinks are relevant

ASReview does not label certain records as relevant, it only provides a ranking of predicted relevancy. This increases the chances for the screener to find (most of) the relevant papers earlier in the screening phase (without going through all 50k records). So you understand correctly that ASReview uses a different approach!

So for your second question, because you do not have prior knowledge on the actual number of relevant papers in your dataset, it is not possible to make any predictions on how much you should still screen. If there was such an option, then it would not be so difficult to determine when to stop screening (the "How to stop?" section in the [active learning explained blogpost](https://asreview.nl/blog/active-learning-explained/).

> e.g., the number of articles in a row that has been included by the researcher?

What do you mean with this exactly?

Let me know if you need any further information!

**Reply by Rensvandeschoot** [reply]
Concerning the second question, you might want to take a look at the Progress chart. 

> It plots the number of relevant records in the last 10 records that you reviewed in ASReview LAB. For example, when you reviewed 100 records, you labeled 3 relevant records between the 91st and 100th reviewed records.

see RTDs: https://asreview.readthedocs.io/en/latest/progress.html#charts

You can also export the data underlying the chart to create different plots or statistics.

**Reply by DLiangUU** [reply]
Hi, 
I see, that makes good sense. I will in that case change my approach to reviewing until I have 50 consecutive irrelevant papers using the tool. In the linked blog post, the switching strategy seems like a good approach to make sure all relevant papers are included. As I’m not that well versed in the different algorithm methods and techniques, which method would be good to switch to after the first round of screening (If I start with the default settings)? Are there any of the methods that are better as the second method than the others?

Regarding “the number of articles in a row that has been included by the researcher”, I was thinking of something similar to the “irrelevant records since last relevant” but for relevant instead (relevant records since last irrelevant). I see Rens replied with what I was thinking of. 

Thank you very much for the information!

**Reply by Rensvandeschoot** [reply]
You might want to read this paper by @JTeijema on [switching](https://asreview.nl/blog/project/switch-models/). Based on his simulation study it is mostly the selection of the feature matrix which contributes most to a performance gain.

**Reply by gimoAI** [reply]
The default settings (TF-IDF with naive Bayes) of ASReview perform well with little training data and therefore make a good starting point. You can check out the paper by @JTeijema mentioned above for switching and there is an example of applying switching in [this paper by Brouwer et al.](https://psyarxiv.com/j6nqz).

If you have around 50.000 articles, I would probably take a somewhat higher number of consecutive irrelevant papers as a stopping rule.

**Reply by DLiangUU** [reply]
Hi both, 

Thank you both very much for the articles and information and sorry for the very slow reply! They were very informative, especially very nice to see the different settings compared with each other as well as the comparison of the different settings in the switching. 

As suggested, I will proceed with the default setting in the first screening and the switch for the second screening round.   

Thank you very much for all your help!
